classes/StaffManager.php

```diff
@@ -0,0 +1,575 @@
+<?php
+
+class StaffManager {
+    public User $player;
+    public System $system;
+
+    public int $staff_level;
+    public int $support_level;
+    public int $user_id;
+    public string $user_name;
+
+    const PERM_BAN_VALUE = -1;
+    const OW_MIN = 10;
+    const OW_MAX = 1000;
+
+    const BAN_TYPE_GAME = 'game';
+    const BAN_TYPE_CHAT = 'tavern';
+    const BAN_TYPE_PM = 'PM';
+    //These ban types do not need to be added to the ban_types array
+    //They are managed in a different manner
+    const BAN_TYPE_AVATAR = 'avatar';
+    const BAN_TYPE_JOURNAL = 'journal';
+    const BAN_TYPE_IP = 'restricted site access';
+    public static array $ban_types = [
+        self::BAN_TYPE_CHAT, self::BAN_TYPE_GAME, self::BAN_TYPE_PM
+    ];
+
+    const VERDICT_UNHANDLED = 0;
+    const VERDICT_GUILTY = 1;
+    const VERDICT_NOT_GUILTY = 2;
+
+    const STAFF_NONE = 0;
+    const STAFF_MODERATOR = 1;
+    const STAFF_HEAD_MODERATOR = 2;
+    const STAFF_CONTENT_ADMIN = 3;
+    const STAFF_ADMINISTRATOR = 4;
+    const STAFF_HEAD_ADMINISTRATOR = 5;
+    public static array $staff_level_names = [
+        self::STAFF_NONE => [
+            'short' =>'User',
+            'long' => 'User',
+        ],
+        self::STAFF_MODERATOR => [
+            'short' => 'Mod',
+            'long' => 'Moderator',
+        ],
+        self::STAFF_HEAD_MODERATOR => [
+            'short' => 'Head Mod',
+            'long' => 'Head Moderator',
+        ],
+        self::STAFF_CONTENT_ADMIN => [
+            'short' => 'Content Admin',
+            'long' => 'Content Administrator',
+        ],
+        self::STAFF_ADMINISTRATOR => [
+            'short' => 'Admin',
+            'long' => 'Administrator',
+        ],
+        self::STAFF_HEAD_ADMINISTRATOR => [
+            'short' => 'Head Admin',
+            'long' => 'Head Administrator',
+        ],
+    ];
+
+    const RECORD_NOT_MIN_SIZE = 20;
+    const RECORD_NOTE = 'note';
+    const RECORD_BAN_ISSUED = 'ban';
+    const RECORD_BAN_REMOVED = 'ban_removed';
+    const RECORD_OFFICIAL_WARNING = 'official_warning';
+
+    const STAFF_LOG_MOD = 'mod_action';
+    const STAFF_LOG_HEAD_MOD = 'head_mod_action';
+    const STAFF_LOG_ADMIN = 'admin_action';
+    const STAFF_LOG_SUPPORT = 'support_action';
+
+    public static array $verdicts = [
+        self::VERDICT_UNHANDLED => 'Unhandled',
+        self::VERDICT_GUILTY => 'Guilty',
+        self::VERDICT_NOT_GUILTY => 'Not Guilty'
+    ];
+
+    public function __construct(User $player) {
+        $this->player = $player;
```

Comment:
In the interest of avoiding a recursive data structure, since this `player` property doesn't seem to actually be used anywhere else in this class, maybe we just store the individual variables and not the instance of the whole class?